src/Components/FilterControls.js

```diff
@@ -1,86 +1,101 @@
 import React, { Component } from 'react';
 import CardContainer from './CardContainer';
+import '../FilterControls.css';
 
 class FilterControls extends Component {
   constructor(props) {
     super(props);
     this.state = {
       location: '',
       time: '',
-      date: ''
+      date: '',
+      matchedIds: []
     }
   }
 
-  getClickedLocation = (event) => {
-    let selectedLocation = event.target.value;
+  getClickedValue = (event) => {
     this.setState({
-      location: selectedLocation
-    }, this.filterByLocation)
+      [event.target.id]: event.target.value
+    }, this.filterParades)
   }
 
-  
-  filterByLocation = () => {
-    return this.props.parades.filter(parade => {
-      return parade.location === this.state.location
-    })
-  }
-  
-  getClickedTime = (event) => {
-    let selectedTime = event.target.value;
-    this.setState({
-      time: selectedTime
-    }, this.filterByTime)
+  filterParades = () => { 
+    const { location, time, date } = this.state;
+    const { parades } = this.props;
+    let matchedIds = [];
+    let matchedObjects = this.returnMatchingObjects();
+
+    parades.forEach(parade => {
+      if (!matchedObjects.includes(parade) && 
+      parade.location === location || 
+      parade.dayOrNight === time ||
+      parade.date === date)
+      {
+        matchedObjects.push(parade);
```

Comment:
What are you actually doing with this `matchedObjects` array? It looks like you create a variable for it, but then it's never actually used anywhere.